_posts/2022-12-23-aws-lambda-snapstart-spring-cloud-function.md

```diff
@@ -0,0 +1,123 @@
+This Lambda function, when invoked, will retrieve some JSON data from a [dummy REST API](https://dummyjson.com/){:target="_blank" rel="noopener noreferrer"} and return it to the user.
+The code can be found on [Github](https://github.com/ordina-jworks/aws-lambda-snapstart-spring-boot){:target="_blank" rel="noopener noreferrer"}.
+
+We made use of the [AWS Serverless Application Model (SAM)](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html){:target="_blank" rel="noopener noreferrer"} to build our Lambda function and deploy it to AWS.
+Enabling the SnapStart feature can be easily done by adding the following two lines to the `Properties` section of the Lambda function resource in the `template.yaml` file used by AWS SAM:
+```
+SnapStart:
+  ApplyOn: PublishedVersions
+```
+
+We started by invoking our Lambda function's unpublished version ($LATEST), in which case SnapStart is not used, and received the following summary from AWS:
+
+<img src="{{ '/img/2022-12-23-aws-lambda-snapstart-spring-cloud-function/lambda-cold-start.png' | prepend: site.baseurl }}" alt="Summary lambda without SnapStart" class="image fit">
+
+We can observe an **Init duration** of around 2.7s, i.e. the time that is spent initializing the execution environment for our Lambda function.
+
+Then, we manually published a new version of our Lambda function using the AWS Console.
+This can done by navigating to the _Versions_ tab of our Lamdba function and pressing the _Publish new version_ button.
+
+<img src="{{ '/img/2022-12-23-aws-lambda-snapstart-spring-cloud-function/lambda-versions.png' | prepend: site.baseurl }}" alt="Lambda function verions" class="image fit">_Versions tab listing all published versions of a Lambda function._
+{: refdef} 
+
+Invoking this newly published version provides us with the following summary:
+
+<img src="{{ '/img/2022-12-23-aws-lambda-snapstart-spring-cloud-function/lambda-snapstart.png' | prepend: site.baseurl }}" alt="Summary lambda with SnapStart" class="image fit">
+
+In this case, we can see SnapStart is used.
+The initialization of the execution environment, represented by the **Init duration** we saw earlier, now happens when publishing the new version.
+Only the restoration of the snapshot, represented by the **Restore duration**, is performed now.
+
+It is quite clear that using Lambda SnapStart is advantageous in most cases.
+We managed to decrease the cold start execution time of our Lambda function from almost 5s (**Init duration** + **duration**) to around 2.6s (**Restore duration** + **duration**), just by enabling this feature.
+
+## Conclusion
+SnapStart is a really great feature and can save a lot of time in your application flow.
+It's a feature that should have been present already as it comes a bit too late. 
```

Comment:
Explain why it's too late?

Additionally in the conclusion: 
- Describe when you'd recommend it (1 sentence) and when not (1 sentence). 
- Add the Java specific context to the conclusion
- What alternatives are there compared to this to prevent cold starts?
- Close with "If you want more information, feel free to reach out ...."